clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -675,7 +675,7 @@ object GarbageCollector {
     val time = DateTimeFormatter.ISO_INSTANT.format(java.time.Clock.systemUTC.instant())
     writeParquetReport(commitsDF, reportLogsDst, time, "commits.parquet")
 
-    val removedCount = removed.count()
+    val removedCount = removed.cache().count()
```

Comment:
This will probably not work where it is most needed.  [Dataset.cache](https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/Dataset.html#cache():Dataset.this.type) persists to "memory and [local] disk".  But I doubt that we have enough disk on the workers to cache the entire dataset.

I believe that the solution is to _write_ first, then re-read and count _that_.